packages/catalog-client/src/CatalogClient.ts

```diff
@@ -92,6 +92,16 @@ export class CatalogClient implements CatalogApi {
     );
 
     const refCompare = (a: Entity, b: Entity) => {
+      // in case field filtering is used, these fields might not be part of the response
```

Comment:
Might not be the most elegant fix, but it works 😄

Reply:
Wanna check `a.kind`, `a.metadata?.namespace` and `a.metadata?.name` right?

Reply:
At least `*.kind` and `*.metadata.name`. Isn't `namespace` an optional field in the catalog? But I'm not sure if it also optional at this place. At least the `stringifyEntityRef` function is able to handle it, but then the order might still be invalid if someone chose to request only the kind and the name.

Reply:
Yeah indeed metadata is still optional (even if there's a default policy that sets it to default)